### PR TITLE
Add conversational entity linking into REL

### PR DESCRIPTION
**Pull request by stefsmeets**
This PR implements [converstational entity linking](https://github.com/informagi/conversational-entity-linking-2022) from @hideaki-j into REL.

First I refactored the CREL code (see https://github.com/informagi/conversational-entity-linking-2022/pull/1). I left the code as-is as much as possible, and tried to depend on functions already in REL where possible. Then I copied all the code over into `./src/REL/crel`.

I added some tests and documentation for the new CREL functions.

I also refactored the server to use fastapi/pydantic (in #147 so you can see the changes). I then branched this off #147, which makes it much easier to implement the conversational entity linking into the server. Note that the API itself did not change, I just added conversational entity linking to it. Fastapi/pydantic will make it much easier to update the API later on.

To check out the server working, see `./scripts/test_server.py` for an example.

### Todo

- [x] Align conv EL with REL itself (re-use parts where possible)
- [x] ConvEL has its own mention detection using Bert, this should be aligned with @eriktks work.
    -> Revisit when #151 is merged
- [x] Figure out what the correct API for interacting with the server should be.
- [x] Update documentation (setup, paths etc, how much should this be part of REL?)

Follow-up:
- Refactor [`bert_md.py`](https://github.com/informagi/REL/blob/fastapi-conv-el/src/REL/crel/bert_md.py) once #151 gets merged.
- Update API (see https://github.com/informagi/REL/issues/146#issuecomment-1375530415)

### CODE REVIEW

**Comment by stefsmeets** [comment]
Hi @chriskamphuis and @hideaki-j I think this PR is pretty much done, could you have a look at it and let me know what you think?

**Comment by chriskamphuis** [comment]
@hideaki-j Can you look at this, as you created CREL, I am not familiar with it

**Comment by hideaki-j** [comment]
I have skimmed through the changes of CREL parts, and it seems to be OK (Stef and I had a meeting about this and I have already shared this with him 😉) I will take the time to check the documentation and test etc in detail when I find time but I think it is good to move to the next step for now.

**Comment by stefsmeets** [comment]
Hi @hideaki-j @chriskamphuis if you are happy with the changes, don't forget to press the 'Approve changes' button, so that the code can be merged :-)

File: src/REL/crel/s2e_pe/utils.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    return predicted_clusters
+
+
+def mask_tensor(t, mask):
+    t = t + ((1.0 - mask.float()) * -10000.0)
+    t = torch.clamp(t, min=-10000.0, max=10000.0)
+    return t
+
+
+# def write_meta_data(output_dir, args):
```
Comment (chriskamphuis) [inline src/REL/crel/s2e_pe/utils.py:139]: Lines after this can be removed I think?

Comment (stefsmeets) [reply src/REL/crel/s2e_pe/utils.py]: Yup, good catch!

**Review by chriskamphuis — APPROVED** [review]

**Review by hideaki-j — APPROVED** [review]

**Pull request merged**